web-console/src/views/query-view/query-view.tsx

```diff
@@ -205,14 +209,23 @@ export class QueryView extends React.PureComponent<QueryViewProps, QueryViewStat
 
         const query = QueryView.isJsonLike(queryString) ? Hjson.parse(queryString) : queryString;
 
+        const queryId = generateQueryId();
+
+        const isSql = !QueryView.isJsonLike(queryString);
+
         let context: Record<string, any> | undefined;
         if (!isEmptyContext(queryContext) || wrapQueryLimit || mandatoryQueryContext) {
           context = { ...queryContext, ...(mandatoryQueryContext || {}) };
+          context.sqlQueryId = queryId;
           if (typeof wrapQueryLimit !== 'undefined') {
             context.sqlOuterLimit = wrapQueryLimit + 1;
           }
         }
 
+        void cancelToken.promise.then(() => {
+          void Api.instance.delete(`/druid/v2/${isSql ? '/sql' : ''}/${queryId}`);
```

Comment:
will `void` here be enough to make sure nothing complains about an unhandled promise rejection (we do not want to trigger https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event)? Can you try this when it fails? Does the error get fully ignored or is there a need for `.catch(() => {})` ? If I was in your place I would add the `.catch(() => {})` just to be sure (and to save time on doing the proper research)

Reply:
yes, you are right, void would not be enough to handle the promise reject. i added catch and throw new DruidError(e). But i did not handle the cancel api response yet.